src/test/ui/nll/user-annotations/normalization-2.rs

```diff
@@ -0,0 +1,114 @@
+// Make sure we honor region constraints when normalizing type annotations.
+
+// check-fail
+
+#![feature(more_qualified_paths)]
+
+trait Trait {
+    type Assoc;
+}
+
+impl<T> Trait for T
+where
+    T: 'static,
+{
+    type Assoc = MyTy<()>;
+}
+
+enum MyTy<T> {
+    Unit,
+    Tuple(),
+    Struct {},
+    Dumb(T),
+}
+
+impl<T> MyTy<T> {
+    fn method<X>() {}
+    fn method2<X>(&self) {}
+}
+
+type Ty<'a> = <&'a () as Trait>::Assoc;
```

Comment:
add a similar test for inherent associated types. cc #105315.
```suggestion
type Ty<'a> = CustomTy::<'a>::Assoc;
```